kedro/config/abstract_config.py

```diff
@@ -26,6 +26,15 @@ def __init__(
         self.env = env
         self.runtime_params = runtime_params or {}
 
+    # From Python 3.12 __getitem__ isn't called in UserDict.get()
+    # Use the version from 3.11 and prior
+    def get(self, key: str, default: Any = None) -> Any:
+        "D.get(k[,d]) -> D[k] if k in D, else d.  d defaults to None."
+        try:
+            return self[key]
+        except KeyError:
+            return default
+
```

Comment:
Why is this needed?

Reply:
Should this be on OmegaConfigLoader or the abstract method? I am slightly confused why `get` method is needed for OmegaConf.

Reply:
> Should this be on OmegaConfigLoader or the abstract method?

I went for the AbstractConfigLoader to catch any cases where users have defined their own Config Loaders (for whatever reason) that inherit from the AbstractConfigLoader. 

> I am slightly confused why get method is needed for OmegaConf.

Whilst we don't promote using get over the dict syntax, we have made it available - and it features in our API docs - allowing users to do something like:
```
conf_loader = OmegaConfigLoader(<...>)
if not conf_loader.get("parameters"):
    parameters = MY_LOCAL_PARAMS
```

Ultimately, removing get() is a breaking change, which would block adding 3.12 support until the next major release.